closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -211,13 +211,14 @@ protected int engineGetOutputSize(int inputLen) {
     @Override
     protected AlgorithmParameters engineGetParameters() {
         AlgorithmParameters params = null;
+        byte[] nonceData = (initialized || nonce != null) ? nonce : createRandomNonce(null);
```

Comment:
What is the motivation for the test of `initialized` here? If it is true and `nonce` is `null` we'll get a `NullPointerException` below. Why don't we write:
```
  byte[] nonceData = (nonce != null) ? nonce : createRandomNonce(null);
```
I have the same question about ibmruntimes/openj9-openjdk-jdk16#13.

Reply:
Same question for ibmruntimes/openj9-openjdk-jdk11#383.

Reply:
@shanchao95 Perhaps you meant to write:
```
byte[] nonceData = (initialized && nonce != null) ? nonce : createRandomNonce(null);
```
so that even if `nonce != null` a random nonce is used after a 'final' operation, like [here](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java#L709)?

Reply:
I opened PRs for Java 11, 16 and 17 with my proposed change.

Reply:
@shanchao95 @ashbm5 can you please take a look at Keith's suggested update, and comment if it looks like what was intended. You can either comment here or in one of the PRs below.

These are all the same change for the different versions:
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/385
https://github.com/ibmruntimes/openj9-openjdk-jdk16/pull/27
https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/288